server/lib/job-queue/handlers/video-upload-tmp-folder-cleaner.ts

```diff
@@ -0,0 +1,53 @@
+import { readdir, unlink, stat } from 'fs-extra'
+import { Stats } from 'node:fs'
+import * as path from 'path'
+import { logger } from '@server/helpers/logger'
+import { getTmpPath } from '@server/helpers/utils'
+import { SCHEDULER_INTERVALS_MS } from '@server/initializers/constants'
+
+async function processVideoUploadTmpFolderCleaner () {
+  const tmpPath = await getTmpPath()
+  let files: string[]
+
+  logger.debug('processVideoUploadTmpFolderCleaner: Reading files from path %s', tmpPath)
+
+  try {
+    files = await readdir(tmpPath)
+  } catch (error) {
+    logger.error('Failed to read resumable video upload tmp folder.', { error })
+    return
+  }
+
+  logger.debug('processVideoUploadTmpFolderCleaner: Found %d files', files.length)
+
+  await Promise.all(files.map(async function (file) {
+    const filePath = path.join(tmpPath, file)
+    let statResult: Stats
+
+    try {
+      statResult = await stat(filePath)
+    } catch (error) {
+      logger.error(`Failed to run stat for ${filePath}`, { error })
+      return
+    }
+
+    // current time (in miliseconds):
+    const now = new Date().getTime()
+    // time of last status change, plus 1h (in miliseconds):
+    const endTime = new Date(statResult.ctime).getTime() + SCHEDULER_INTERVALS_MS.removeTmpVideo
```

Comment:
I don't think you should use `SCHEDULER_INTERVALS_MS.removeTmpVideo`

Instead, use a private variable to store the last run time (so do nothing for the first run), and delete files < `lastRunTime`

Also, I don't think 1 hour is enough since uploads can be very long